### DISCUSSION TITLE
[Question] Plans for "desktop-first" controls?

### DISCUSSION THREAD

**Discussion by trympet** [dotnet/maui#1590] [Ideas]
I attempted to rewrite my production WPF application to MAUI to get a taste of whats to come, since I have plans publish the application to macOS. Sadly, the takeaway from my experience was that MAUI is not the correct framework for me; the lack of desktop-class controls, such as DataGrid, GridView, context menus, etc. was a deal-breaker.

From what I can tell, MAUI will not bring any such desktop-like controls. Therefore, it seems that MAUI is a mobile UI framework, which happens to run on desktop :/

My question is: are there plans to add desktop-esc controls, such as DataGrid? Is writing desktop-only or desktop-first applications even a supported use case?

Is Blazor Desktop the project which I should focus my attention to instead? I know HTML5 has all the controls I need. If so, this is confusing, since, from what I recall, Microsoft itself said that Blazor Desktop is a mobile-first framework. Maybe Avalonia instead?

**Comment by trympet** [comment]
The scope of .NET MAUI should be clear and apparent when you visit the docs. First-party desktop controls aren't coming, and MAUI isn't ready for LoB apps, or a replacement for WPF.

MAUI should be promoted as a replacement for UWP instead.

...oh wait, we have WinUI for that 😕

**Reply by u2m4c6** [reply]
I think this deserves a reply. Mobile first controls on desktop isn't anything more than putting lipstick on Xamarin. Which is fine I guess, but then just call it Xamarin 6.

**Reply by Arkenar** [reply]
So i shouldn't use .NET MAUI for a Desktop-first application?
